server/schedule/checker/rule_checker.go

```diff
@@ -111,8 +111,15 @@ func (c *RuleChecker) fixRulePeer(region *core.RegionInfo, fit *placement.Region
 
 func (c *RuleChecker) addRulePeer(region *core.RegionInfo, rf *placement.RuleFit) (*operator.Operator, error) {
 	checkerCounter.WithLabelValues("rule_checker", "add-rule-peer").Inc()
-	ruleStores := c.getRuleFitStores(rf)
-	store := c.strategy(region, rf.Rule).SelectStoreToAdd(ruleStores)
+	var (
+		strategy     = c.strategy(region, rf.Rule)
+		ruleStores   = c.getRuleFitStores(rf)
+		extraFilters = make([]filter.Filter, 0)
+	)
+	if len(strategy.locationLabels) > 0 && len(strategy.isolationLevel) > 0 {
+		extraFilters = append(extraFilters, filter.NewIsolationFilter(strategy.checkerName, strategy.isolationLevel, strategy.locationLabels, ruleStores))
+	}
```

Comment:
This block only uses variables of `strategy`. Better move to `strategy.SelectToAdd`.